web/pdf_history.js

```diff
@@ -75,38 +75,74 @@ var PDFHistory = (function () {
 
       var self = this;
       window.addEventListener('popstate', function pdfHistoryPopstate(evt) {
-        evt.preventDefault();
-        evt.stopPropagation();
-
         if (!self.historyUnlocked) {
           return;
         }
-        if (evt.state) {
+        if (self._isStateObjectDefined(evt.state)) {
```

Comment:
Please revert this change, since there's an edge-case where it can can have undesired effects.
In the `GENERIC` viewer (which support opening multiple PDF files), `evt.state` may refer to a previously opened PDF file. In that case, nothing should be done; note how [`_goTo`](https://github.com/mozilla/pdf.js/blob/master/web/pdf_history.js#L346-L350) returns early for this case (given `self._isStateObjectDefined(evt.state) == false`).

If this is changed, we could incorrectly reach the code below, which should *only* happen when the user modifies the hash.